src/commands/command.ts

```diff
@@ -492,15 +558,25 @@ export class CommandsManager {
 
   /** Add a Command */
   add(cmd: Command | typeof Command): boolean {
+    let CmdClass: typeof Command | undefined
     if (!(cmd instanceof Command)) {
-      const CmdClass = cmd
+      CmdClass = cmd
       cmd = new CmdClass()
       Object.assign(cmd, CmdClass.meta ?? {})
     }
     if (this.exists(cmd, cmd.extension?.subPrefix))
       throw new Error(
         `Failed to add Command '${cmd.toString()}' with name/alias already exists.`
       )
+    if (cmd.name === '' && CmdClass !== undefined) {
+      let name = CmdClass.name
+      if (
+        name.toLowerCase().endsWith('command') &&
+        name.toLowerCase() !== 'command'
+      )
+        name = name.substr(0, name.length - 'command'.length).trim()
+      cmd.name = name
+    }
     if (cmd.name === '') throw new Error('Command has no name')
```

Comment:
I think `!cmd.length` looks better than `cmd.name === ''`

Reply:
Linter will compain on `!cmd.length` and say to use `cmd.length === 0` 🤔

Reply:
Oops, Okay